core/src/credential.rs

```diff
@@ -55,6 +55,20 @@ pub struct Credential {
     pub property_set: HashMap<String, Value>,
 }
 
+#[derive(Debug, Serialize, Deserialize, Clone)]
+pub struct IntegrityProof {
```

Comment:
We should replace `Identityproof` with the `DataIntegrityProof` object in the `proof.rs` file. Let's make sure we just have one source of truth on this